pandas/core/ops/__init__.py

```diff
@@ -157,13 +157,22 @@ def maybe_upcast_for_op(obj):
         # implementation; otherwise operation against numeric-dtype
         # raises TypeError
         return Timedelta(obj)
-    elif isinstance(obj, np.timedelta64) and not isna(obj):
+    elif isinstance(obj, np.timedelta64):
+        if isna(obj):
+            # wrapping timedelta64("NaT") in Timedelta returns NaT,
+            #  which would incorrectly be treated as a datetime-NaT, so
+            #  we broadcast and wrap in a Series
+            right = np.broadcast_to(obj, shape)
+
+            # Note: we use Series instead of TimedeltaIndex to avoid having
+            #  to worry about catching NullFrequencyError.
+            return pd.Series(right)
```

Comment:
rather than this can you make a safe method that deals with null frequency, this is a bit odd

Reply:
This basically is the safe method to deal with null frequency.  I think there will be room to simplify it after we merge dispatch_to_index_op into dispatch_to_extension_op

(I agree that I'd prefer these methods not need to know about Series)

Reply:
ok fair enough, maybe make an issue about this? (or just add to your list) :->